spec/services/subscriptions/override_service_spec.rb

```diff
@@ -120,4 +121,119 @@
       end
     end
   end
+
+  describe '.call_from_api' do
+    let(:external_customer_id) { customer.external_id }
+
+    let(:plan_params) do
+      {
+        name: plan_name,
+        code: "#{plan.code}-123456",
+        interval: plan.interval,
+        overridden_plan_id: plan.id,
+        pay_in_advance: plan.pay_in_advance,
+        organization_id: organization.id,
+        amount_cents: 299,
+        amount_currency: plan.amount_currency,
+        charges: [
+          {
+            billable_metric_id: billable_metric.id,
+            charge_model: 'standard',
+            properties: {
+              amount: '100',
+            },
+          },
+        ],
+      }
+    end
+
+    let(:subscription_params) do
+      {
+        external_customer_id: external_customer_id,
+        name: 'subscription display name',
+        external_id: SecureRandom.uuid,
+        billing_time: billing_time,
+      }
+    end
+
+    before { plan }
+
+    it 'creates a subscription' do
+      result = override_service.call_from_api(
+        organization: organization,
+        plan_args: plan_params,
+        subscription_args: subscription_params
+      )
+
+      expect(result).to be_success
+
+      subscription = result.subscription
+
+      aggregate_failures do
+        expect(subscription.customer_id).to eq(customer.id)
+        expect(subscription.plan_id).not_to eq(plan.id)
+        expect(subscription.started_at).to be_present
+        expect(subscription.subscription_date).to be_present
+        expect(subscription).to be_active
+        expect(subscription).to be_anniversary
+      end
+    end
+
+    it 'creates a new plan' do
+      expect do
+        override_service.call_from_api(
+          organization: organization,
+          plan_args: plan_params,
+          subscription_args: subscription_params
+        )
+      end.to change(Plan, :count).by(1)
+    end
+
+    it 'creates a new plan that is correctly linked to overridden_plan' do
+      result = override_service.call_from_api(
+        organization: organization,
+        plan_args: plan_params,
+        subscription_args: subscription_params
+      )
+
+      expect(result.subscription.plan.overridden_plan_id).to eq(plan.id)
+    end
+
+    context 'when plan is invalid' do
+      let(:plan_name) { nil }
+
+      it 'fails' do
+        result = override_service.call_from_api(
+          organization: organization,
+          plan_args: plan_params,
+          subscription_args: subscription_params
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)